### ISSUE TITLE
`cloup.Group` ignores `command_class` to produce sub-commands

### ISSUE THREAD

**Issue by kdeldycke** [janluke/cloup#160]
Using:
- Cloup v2.1.1
- Click v8.1.4
- Python 3.11.4

#### Bug description

`cloup.Group` ignore the [`command_class` property](https://github.com/pallets/click/blob/d9af5cfa009c927a96d10ed38a3e37979876a12e/src/click/core.py#L1797-L1802) that is [used in `click.Group` to set the default class of the `@group.command()` decorator](https://github.com/pallets/click/blob/d9af5cfa009c927a96d10ed38a3e37979876a12e/src/click/core.py#L1855-L1894).

#### To Reproduce

Here is the minimal CLI, saved in a `cloup_test.py` file, that is reproducing the issue:

```python
from cloup import Command, Group, group


class CustomCommand(Command):

    def __init__(self, *args, **kwargs):
        kwargs.setdefault("context_settings", {"help_option_names": ("--help", "--my-fancy-help")})
        super().__init__(*args, **kwargs)


class CustomGroup(Group):

    command_class = CustomCommand


@group(cls=CustomGroup)
def my_cli():
    pass


@my_cli.command()
def subcommand():
    pass


if __name__ == "__main__":
    my_cli()
```

When I call the bare CLI at the group level, the `subcommand` is properly registered and appears in the help screen:
```shell-session
$ python ./cloup_test.py 
Usage: cloup_test.py [OPTIONS] COMMAND [ARGS]...

Options:
  --help  Show this message and exit.

Commands:
  subcommand
```

Now when I call the `--help` on the `subcommand` I cannot see any reference to my `--my-fancy-help` custom help option name:

```shell-session
$ python ./cloup_test.py subcommand --help
Usage: cloup_test.py subcommand [OPTIONS]

Options:
  --help  Show this message and exit.
```

#### Expected behavior

Instead of the output above, I expect to get the following results:

```shell-session
$ python ./cloup_test.py subcommand --help
Usage: cloup_test.py subcommand [OPTIONS]

Options:
  --help, --my-fancy-help  Show this message and exit.
```

Notice how `--my-fancy-help` is featured in the help screen, because I expect `@my_cli.command()` to return a decorator of `CustomCommand`, as per the `command_class` property defined on `CustomGroup`.

#### Click behavior

The same minimal CLI, sourced with Click's primitives, is working as expected.

In the example above, if you replace:
```python
from cloup import Command, Group, group
```

With:
```python
from click import Command, Group, group
```

You get the expected output:
```shell-session
$ python ./cloup_test.py subcommand --help
Usage: cloup_test.py subcommand [OPTIONS]

Options:
  --my-fancy-help, --help  Show this message and exit.
```

#### Notes

- This might be related to https://github.com/pallets/click/pull/2417 , which has been fixed in the recent Click 8.1.4

- This example is [inspired by Click's unittest](https://github.com/pallets/click/blob/d9af5cfa009c927a96d10ed38a3e37979876a12e/tests/test_command_decorators.py#L14-L31)

**Comment by janluke** [comment]
Thanks for reporting. I fixed it but I'm having issues with mypy.

**Comment by kdeldycke** [comment]
> Thanks for reporting. I fixed it but I'm having issues with mypy.

Thanks @janluke for the quick update! You're fast!

As for Mypy I guess you shouldn't try to be too perfect. You're better than me at types, but you have to realize the typing you're doing in Cloup is quite advanced. Click itself has some difficulties with typing the latest 8.1.4 release. See: https://github.com/pallets/click/issues/2558

What I wanted to say: it is OK for me to have a couple of `# type: ignore[]` in the code if it makes your life easier as a maintainer. You'll have time to revisit these issues and perfect it in next releases.

**Comment by kdeldycke** [comment]
Ah, and see @davidism comment [about being reasonable with typing in Click](https://github.com/pallets/click/issues/2558#issuecomment-1624407433): 

> Also, note that typing in Python is optional. Click is not broken by this, 8.1.4 works just fine and fixes a lot of issues. You have the power to ignore any typing errors that are not useful, either inline, per file, or across a project if you need to get CI to pass. It's ok, you're not suddenly going to write the decorator wrong 😉.

**Comment by janluke** [comment]
I've always considered typing (and good IDE autocompletion) a feature of Cloup. If I use `cloup.group`, then I expect my IDE knows I can call `section()`. Unfortunately, Click is not designed with typing in mind.

**Comment by janluke** [comment]
Actually, even `master` fails mypy checks now: https://github.com/janluke/cloup/actions/runs/5542413108/jobs/10117062431

**Comment by kdeldycke** [comment]
> Actually, even `master` fails mypy checks now: https://github.com/janluke/cloup/actions/runs/5542413108/jobs/10117062431

Yeah probably because of `click==8.1.4`.

To catch regression as fast as possible and reduce the maintenance overhead, I:
- [Run tests every week](https://github.com/kdeldycke/click-extra/blob/6334a2815b82f71e8e0d855dfd48d6b7aded3597/.github/workflows/tests.yaml#L8-L10)
- Test [against Click `8.1.x` branch and `main` (i.e. the upcoming `8.2.x` branch)](https://github.com/kdeldycke/click-extra/blob/6334a2815b82f71e8e0d855dfd48d6b7aded3597/.github/workflows/tests.yaml#L59-L60)
- Test [against Cloup stable and `master` release](https://github.com/kdeldycke/click-extra/blob/6334a2815b82f71e8e0d855dfd48d6b7aded3597/.github/workflows/tests.yaml#L62-L65)

That way I can fix or report issues quickly, before they compounds.

**Comment by janluke** [comment]
I checked the click issue only a few minutes ago, I hadn't understood that click 8.1.4 was the cause of the issue. So, I'll fix this issue when I have time, maybe after work today.

Yes, I'll add weekly tests as well.

**Comment by kdeldycke** [comment]
I think your issues are being addressed by Click `8.1.5` which was released 40 minutes ago.

**Comment by kdeldycke** [comment]
I can confirm the latest release of Cloup v2.1.2 is addressing this issue.

Thanks again @janluke for the fast release!
